### ISSUE TITLE
Add DDN-informed BOM Summary section without generating an actual BOM

### ISSUE THREAD

**Issue by jwsiejk** [jwsiejk/accountlist#367]
## Goal
Add a **BOM Summary** capability to the AI Workload Discovery Mapper that explains, in plain English, how the selected workload and captured inputs map to DDN-informed architecture considerations — without creating an actual BOM, exact sizing, SKU list, or quote.

## Important intent
The Workload Summary explains the workload in plain English for a sales counterpart.

The new **BOM Summary** should be the next deeper layer:

> Given this workload and the filled discovery/BOM fields, what architecture areas matter and why?

It should use DDN-informed reference architecture/use-case language wherever we have source-backed information, but it must not invent official DDN sizing guidance.

## Critical guardrail
Do **not** generate an actual BOM.
Do **not** output exact node counts, exact GPU counts, exact DDN SKUs, pricing, or final sizing recommendations.
Do **not** claim something is official DDN guidance unless the repo contains source-backed DDN reference content for it.

Use phrasing like:

```text
This points toward...
This should be validated through sizing and benchmarking...
A DDN-style architecture discussion would focus on...
The next step is to confirm throughput, metadata profile, GPU count, and resiliency requirements before a BOM.
```

## Source-backed DDN information requirement
The app should have a clean data structure for DDN-informed architecture notes.

If the repo already contains DDN reference architecture/use-case content, use it.
If it does not, create a clearly marked curated data file with **source notes/placeholders** and conservative language.

Preferred structure:

```text
ui/partner-hub/components/workload-mapper/workload-mapper-ddn-reference.ts
```

or similar.

Each workload should include DDN-informed sections such as:

- Workload pattern
- DDN-relevant architecture focus
- Data platform considerations
- Compute/GPU considerations
- Networking / fabric considerations
- Performance validation questions
- Governance / security considerations
- What must be confirmed before a real BOM
- Plain-English explanation

## Workloads to cover
Cover all preloaded workloads:

### FSI
- Fraud Detection & Investigation
- Risk Modeling / Stress Testing
- Trading / Quant Research
- Compliance / Document Intelligence
- RAG Knowledge Assistant
- Model Training / Fine-tuning

### HPC / AI
- Large-scale AI Training
- Inference Platform
- Scientific Simulation
- Genomics / Life Sciences
- Media / Rendering Pipeline

### Custom
- Provide a generic custom workload BOM Summary pattern based on selected AI pattern and captured fields.

## DDN-informed architecture themes to use carefully
Use these themes where appropriate, but keep them as architecture considerations, not final BOM claims:

### AI training / fine-tuning
- GPU utilization depends heavily on keeping data close and feeding accelerators efficiently.
- High-throughput training storage matters for dataset streaming and preprocessing.
- Checkpointing can become a major performance and recovery consideration.
- Distributed training requires attention to data movement, network/fabric, and benchmark validation.
- Final sizing depends on model size, dataset size, batch size, training window, checkpoint frequency, and GPU topology.

### RAG / document intelligence
- Data ingest, chunking, embedding, and retrieval freshness matter.
- Many-small-file/document patterns can stress metadata and indexing workflows.
- Vector DB/retrieval layers need governed access to source content and embeddings.
- Final design depends on corpus size, update frequency, chunk count, query concurrency, latency targets, and access-control model.

### FSI fraud / risk / trading
- Regulated data requires governance, auditability, encryption, retention, and access controls.
- Intraday/near-real-time workflows require faster ingest and retrieval paths than batch-only analytics.
- Quant/trading research depends on fast access to time-series/feature data and high concurrency from researchers/jobs.
- Risk/stress testing depends on reproducibility, scenario outputs, large historical datasets, and parallel data access.

### HPC simulation
- Scheduler, high-speed interconnect, parallel file/data platform, checkpoint/output handling, and post-processing workflows matter.
- Final design depends on job profile, concurrency, file sizes, checkpoint cadence, output volume, and visualization/post-processing needs.

### Inference
- Latency SLOs, concurrency, model serving tier, caching/session state, observability, and guardrails matter.
- Final design depends on model size, request rate, p95/p99 latency target, token volume, context/retrieval needs, and HA design.

## UI requirements
Add a **BOM Summary** card/section near the existing Summarize/Talk Track/BOM Readiness outputs.

Include:

- Button: `Generate BOM Summary`
- Generated output card title: `BOM Summary`
- Clear disclaimer:

```text
This is an architecture/BOM-readiness summary, not an actual BOM, quote, or official sizing recommendation.
```

- Optional Copy button if straightforward.

## Generation approach
Use the same local Ollama-backed pattern already used by the Workload Summary, but with a separate prompt/route or separate mode.

Preferred:

- Keep Workload Summary and BOM Summary separate.
- Add a separate prompt helper such as:

```text
ui/partner-hub/lib/workload-mapper/bom-summary-prompt.ts
```

- Add a separate API route or mode if clean:

```text
/api/workload-mapper/bom-summary
```

or use the existing summarize route only if it remains clean and clearly separated by request type.

## BOM Summary output style
The output should be plain English and structured enough to be useful.

Suggested format:

```text
This is not a final BOM. It is a BOM-readiness summary that explains what the architecture conversation should focus on.

For this workload, the architecture discussion should focus on...

The data platform needs to support...

The compute/GPU conversation should focus on...

The networking/fabric discussion should consider...

The governance/security requirements are...

Before creating a real BOM, we still need to confirm...
```

Do not make it too long. Aim for 4–6 concise sections or paragraphs.

## Use captured fields directly
Use the current tool inputs:

- selected workload/category/description
- AI pattern
- data types
- data size range
- daily ingest range
- file/object pattern
- freshness requirement
- performance tier
- latency requirement
- concurrency
- GPU dependency
- governance fields
- BOM input capture fields
- known/missing inputs
- architecture pipeline
- building blocks
- DDN-informed reference notes

## Missing fields behavior
If BOM-related fields are missing, explain why they matter.
If no required BOM inputs are missing, say:

```text
The required BOM-readiness inputs are currently filled in. The next step is validation with DDN/reference architecture sizing guidance, benchmark expectations, and customer-specific constraints.
```

Do not invent missing fields that the tool does not ask for.
If an additional detail would be useful but is not captured in the tool, phrase it as a future enhancement or deeper architecture review item — not as a missing input.

## Code quality
- TypeScript strongly typed
- No `any`
- Keep DDN-informed reference data in a clean maintainable structure
- No external cloud API calls
- Use local Ollama through server route, same privacy model as Workload Summary
- Preserve readable JSX
- Do not reintroduce CHIP-8 or Steam Trains references

## Validation
Run from the correct package directory:

```bash
npm run lint
npm run typecheck
npm test -- --runInBand
npm run build
rg -i "chip-?8|steam[- ]?trains" . || true
```

If unavailable, document closest equivalent.

## Manual validation
1. Start Ollama locally.
2. Open `/tools/workload-mapper`.
3. Select an FSI workload.
4. Auto-populate realistic example.
5. Generate Workload Summary.
6. Generate BOM Summary.
7. Confirm BOM Summary uses DDN-informed architecture language but does not output exact SKUs/counts/BOM.
8. Confirm output is plain English and useful for a sales/SE architecture conversation.

## Acceptance criteria
- BOM Summary button exists.
- BOM Summary is separate from Workload Summary.
- BOM Summary uses DDN-informed architecture themes/data where available.
- BOM Summary does not create an actual BOM.
- BOM Summary does not invent official DDN sizing/SKU recommendations.
- Output is plain English and detailed enough to guide next architecture/BOM discussion.
- Missing-field behavior is accurate.
- Works for preloaded and custom workloads.
- No external cloud API calls.
- Typecheck/lint/build pass or unavailable commands documented.
- No stale CHIP-8 / Steam Trains references.

**Comment by jwsiejk** [comment]
Additional clarification from James:

The BOM Summary should get as specific as possible **without creating an actual BOM**. The intent is not just to say “this maps to compute/storage/network/security.” It should explain the likely DDN-style/reference-architecture building blocks and what each building block does for the workload.

Important nuance:
- The BOM Input Capture fields may already be filled out.
- The BOM Summary should use those captured fields to explain why the workload maps to certain building blocks.
- The output should not ask for the same fields again if they are already filled in.
- Missing fields should only be discussed if the tool flags them as missing.

Desired BOM Summary behavior:

```text
Based on the captured inputs, this workload maps to these DDN-informed architecture building blocks:

1. High-performance data platform / active data layer
Why: The workload has 52 TB active data, 0.8–1.6 TB/day ingest, and ~6B transaction rows plus 12M case artifacts. That points to a need for fast ingest, high-throughput reads, and metadata/indexing efficiency.

2. Ingestion, indexing, and retrieval pipeline
Why: Fraud teams need near-real-time scoring and investigator context refreshed every 5–15 minutes. That means the architecture must keep new transaction/case data searchable and usable quickly.

3. GPU-backed inference / AI services layer
Why: The captured inputs call for an 8B–13B reasoning assistant, anomaly models, and GPU inference for scoring and summarization. The summary should explain the role of the accelerator layer without estimating GPU count.

4. Governance and security layer
Why: This is regulated financial data with strict audit trail, encryption, regional residency, explainability, RBAC/ABAC controls, and 7-year retention.

5. HA/DR and replication layer
Why: The captured inputs call for active-active scoring endpoints and cross-region replication for governed case data.
```

The summary should explain these building blocks in plain English and tie them directly to the captured inputs.

Guardrails remain:
- Do not output exact DDN SKUs.
- Do not output exact node counts.
- Do not output exact GPU counts.
- Do not create a BOM or quote.
- Do not claim official DDN sizing unless source-backed content exists in the repo.
- If source-backed DDN reference architecture details exist in the repo, use them directly.
- If not, use conservative wording like “DDN-informed,” “DDN-style architecture discussion,” “reference-architecture pattern,” or “should be validated against DDN/reference architecture guidance.”

Preferred wording:

```text
This points toward...
This building block matters because...
The captured inputs suggest...
This should be validated through DDN/reference architecture sizing guidance and benchmark expectations before a real BOM.
```

The BOM Summary should be detailed enough to help James explain the architecture conversation, but not so specific that it becomes an unofficial BOM.
